designdocs/graph_sdk.md

```diff
@@ -0,0 +1,178 @@
+# Graph SDK
+
+## Context and Scope
+
+The proposed design consists of changes that is going to be a separate repository to facilitate
+graph related interactions with Frequency chain in test and production environments
+
+## Problem Statement
+
+Implementing DSNP protocol on Frequency comes with its own challenges and tradeoffs. One of these
+challenges is to decide what is the boundary between DSNP and Frequency and what should be
+implemented directly in Frequency.
+
+To keep Frequency implementation as DSNP agnostic as possible we decided to store social graph
+related data as blobs in Frequency and keep the details of how these blobs are created and modified
+inside DSNP spec. Graph SDK is an implementation of mentioned DSNP spec for social graph, optimized
+for storage and interaction with Frequency.
+## Goals
+
+- Define operations
+- Define interface of Graph SDK.
+- Define the main concepts and entities in Graph SDK.
+- Define the algorithms to optimize the output regarding Frequency.
+
+## Operations
+Following is a list of desired operations in this SDK:
+
+| Name  | Description |
+| ------------- | ------------- |
+| **Initialise**  | Creates in memory graph structure for a desired MSA.  |
+| **Import**  | Import the blob from frequency into Graph SDK for desired MSA.  |
+| **Update**  | Changes the current in-memory graph structure with incoming updates.  |
+| **Get Graph**  | Exposes current state of in-memory graph to the consumer of the SDK.  |
+| **Has Updates**  | Determines if the applied changes created some updates in graph that needs to be persisted on Frequency.  |
+| **Calculate Updates**  | Applies the graph changes and generates optimized blobs to be applied to Frequency  |
+| **Persist**  | This should be called after successful update of Frequency to remove tracking of persisted updates.  |
```

Comment:
Did not really get the idea behind persistence, needs a bit rewording here

Reply:
I agree. Perhaps we need an additional paragraph before this to describe the SDK-in-memory <-> persistence-on-chain model.

Reply:
I'm removing the **persist** for now to simplify the flow. We might need to add it later based on how providers would prefer to interact with the sdk.